postHandler.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const fs = require('fs');
+const querystring = require('querystring');
+const makeTemplate = require('./elementTemplate');
+const updateIndex = require('./indexTemplate');
+
+const postHandler = (req, res) => {
+  console.log(`req.url is: ${req.url}`);
+
+  let body = {};
+
+  req.on('data', (chunk) => {
+    const queriedChunk = querystring.parse(chunk.toString());
+
+    body = queriedChunk;
+  });
+
+  req.on('end', () => {
+
+    fs.readdir('./public', 'utf8', (err, files) => {
+      if (!(files.includes(req.url.slice(1)))) {
+        const data = makeTemplate(body.elementName, body.elementSymbol, body.elementAtomicNumber, body.elementDescription);
+
+        fs.writeFile(`./public/${req.url.slice(1)}`, data, (err) => {
+          if (err) { throw err; }
+        });
+
+        updateIndex(req.url.slice(1), body.elementName);
```

Comment:
All the code here up to the else should be in the writeFile above. That way the index isn't updated until the file is successfully written.

Reply:
Also the updateIndex is also asynchronous, which means that it could finish at any time. You want to send a status 200 okay back to the client after you've finished writing the new element and after you've updated the index. The way it currently is, you'll write the file, update the index, and reply back to the user at the same moment. If any of those fail, you wouldn't be able to notify the user.